static/js/services/http-wrapper.js

```diff
@@ -18,11 +18,16 @@ var _ = require('underscore');
             return p.url !== url;
           });
         },
-        cancelAll: function() {
+        cancelExceptFor: function(allow) {
+          var stillPending = [];
           _.each(pending, function(p) {
-            p.canceller.resolve();
+            if(_.contains(allow, p.updateTarget)) {
+              stillPending.push(p);
+            } else {
+              p.canceller.reject();
```

Comment:
Agreed - calling `reject()` leads to the request continuing, and the success handler being called!  I will add handling to check for cancelled requests, and prevent both success and failure processing.